models/classes/user/OntologyLtiUserService.php

```diff
@@ -79,8 +79,16 @@ public function findUser(\taoLti_models_classes_LtiLaunchData $ltiContext)
             );
 
             $roles = $this->determineTaoRoles($ltiContext);
-
-            $ltiUser = new LtiUser($ltiContext, $instance->getUri(), $properties[PROPERTY_USER_ROLES], (string)current($properties[PROPERTY_USER_UILG]), (string)current($properties[PROPERTY_USER_FIRSTNAME]), (string)current($properties[PROPERTY_USER_LASTNAME]), (string)current($properties[PROPERTY_USER_MAIL]));
+            $lang = current($properties[PROPERTY_USER_UILG]);
+            
+            // In case of the language is a Language Resource in Ontology, get its code.
+            if ($lang instanceof \core_kernel_classes_Resource) {
+                $lang = \tao_models_classes_LanguageService::getCode($lang);
```

Comment:
it should be `\tao_models_classes_LanguageService::singleton()->getCode($lang)` ?